### PR TITLE
[OAP-CACHE][POAE7-472]Branch 0.8 spark 2.4.x external db doc

### PR DESCRIPTION
**Pull request by offthewall123**
## What changes were proposed in this pull request?

update user doc for external DB.

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

**Comment by offthewall123** [comment]
@jikunshang @winningsix @yma11 @yeyuqiang  Please take a review thanks.

File: oap-cache/oap/docs/User-Guide.md [hunk 1]
```diff
@@ -715,3 +716,14 @@ spark.sql("SELECT column_B FROM table_B")
 spark.sql("SET spark.sql.oap.cache.enabled=true")
 spark.sql("SELECT column_C FROM table_C")
 ```
+
+### Large Scale and Heterogeneous cluster support
+To support heterogeneous cluster(some nodes with Optane™ DC Persistent Memory, some without), we introduce an external DB to store cache meta info. Now support [Redis]("https://redis.io/"). Users need to [launch a redis-server]("https://www.runoob.com/redis/redis-install.html").
```
Comment (jikunshang) [inline oap-cache/oap/docs/User-Guide.md:721]: `launch a redis-server` use an official link?

Comment (jikunshang) [inline oap-cache/oap/docs/User-Guide.md:721]: I don't think this part is clear for end users. Do you have any feedback? @winningsix @yeyuqiang @yma11

**Review by jikunshang — CHANGES_REQUESTED** [review]

File: oap-cache/oap/docs/User-Guide.md [hunk 2]
```diff
@@ -715,3 +716,14 @@ spark.sql("SELECT column_B FROM table_B")
 spark.sql("SET spark.sql.oap.cache.enabled=true")
 spark.sql("SELECT column_C FROM table_C")
 ```
+
+### Large Scale and Heterogeneous cluster support
+To support large scale cluster and heterogeneous cluster(some nodes with Optane™ DC Persistent Memory, some without), we introduce an external DB to store cache meta info. Now support [Redis]("https://redis.io/"). Users need to [download and launch a redis-server]("https://redis.io/download").
+External DB stores cache's locality info, it will influence spark to schedule tasks according to cache locality. If there's no cache available, spark will fallback to schedule respecting HDFS locality.
```
Comment (yma11) [inline oap-cache/oap/docs/User-Guide.md:722]: "Now support [Redis]("https://redis.io/"). Users need to [download and launch a redis-server]("https://redis.io/download"). " put this sentence at end of this paragraph and write it like "Currently we support Redis to provide the service and to launch a Redis server, please refer the [official guide]("https://redis.io/download"). "

File: oap-cache/oap/docs/User-Guide.md [hunk 3]
```diff
@@ -417,6 +417,7 @@ When all the queries are done, you will see the `result.json` file in the curren
 - [Cache Hot Tables](#Cache-Hot-Tables)  Data Source Cache also supports caching specific tables according to actual situations, these tables are usually hot tables.
 - [Column Vector Cache](#Column-Vector-Cache)  This document above use **binary** cache as example, if your cluster memory resources is abundant enough, you can choose ColumnVector data cache instead of binary cache to spare computation time.
 - [Disable Cache in Runtime](#Disable-Cache-in-Runtime) Data Source Cache also supports disabling Cache in Runtime.
+- [Large Scale and Heterogeneous cluster support](#Large Scale and Heterogeneous cluster support)
 ### Additional Cache Strategies
```
Comment (yma11) [inline oap-cache/oap/docs/User-Guide.md:421]: Highlight this feature works only with "external" cache strategy?

File: oap-cache/oap/docs/User-Guide.md (hunk 2 above)
Comment (offthewall123) [reply oap-cache/oap/docs/User-Guide.md]: Modified, thanks.

File: oap-cache/oap/docs/User-Guide.md (hunk 3 above)
Comment (offthewall123) [reply oap-cache/oap/docs/User-Guide.md:421]: Yes, it should be added.

File: oap-cache/oap/docs/User-Guide.md (hunk 1 above)
Comment (offthewall123) [reply oap-cache/oap/docs/User-Guide.md]: Yes, have changed it.

**Comment by jikunshang** [comment]
LGTM.

**Comment by yma11** [comment]
LGTM+1.

**Pull request merged**